exporter/xray/segment.go

```diff
@@ -0,0 +1,319 @@
+)
+
+const (
+	traceIDLength    = 35 // fixed length of aws trace id
+	spanIDLength     = 16 // fixed length of aws span id
+	epochOffset      = 2  // offset of epoch secs
+	identifierOffset = 11 // offset of identifier within traceID
+)
+
+type segment struct {
+	// ID - A 64-bit identifier for the segment, unique among segments in the same trace,
+	// in 16 hexadecimal digits.
+	ID string `json:"id"`
+
+	// Name - The logical name of the service that handled the request, up to 200 characters.
+	// For example, your application's name or domain name. Names can contain Unicode
+	// letters, numbers, and whitespace, and the following symbols: _, ., :, /, %, &, #, =,
+	// +, \, -, @
+	Name string `json:"name,omitempty"`
+
+	// StartTime - number that is the time the segment was created, in floating point seconds
+	// in epoch time.. For example, 1480615200.010 or 1.480615200010E9. Use as many decimal
+	// places as you need. Microsecond resolution is recommended when available.
+	StartTime float64 `json:"start_time"`
+
+	// TraceID - A unique identifier that connects all segments and subsegments originating
+	// from a single client request.
+	//	* The version number, that is, 1.
+	//	* The time of the original request, in Unix epoch time, in 8 hexadecimal digits.
+	//	* For example, 10:00AM December 2nd, 2016 PST in epoch time is 1480615200 seconds, or 58406520 in hexadecimal.
+	//	* A 96-bit identifier for the trace, globally unique, in 24 hexadecimal digits.
+	TraceID string `json:"trace_id,omitempty"`
+
+	// EndTime - number that is the time the segment was closed. For example, 1480615200.090
+	// or 1.480615200090E9. Specify either an end_time or in_progress.
+	EndTime float64 `json:"end_time"`
+
+	/* ---------------------------------------------------- */
+
+	// Service - An object with information about your application.
+	//Service service `json:"service,omitempty"`
+
+	// User - A string that identifies the user who sent the request.
+	//User string `json:"user,omitempty"`
+
+	// Origin - The type of AWS resource running your application.
+	Origin string `json:"origin,omitempty"`
+
+	// ParentID – A subsegment ID you specify if the request originated from an instrumented
+	// application. The X-Ray SDK adds the parent subsegment ID to the tracing header for
+	// downstream HTTP calls.
+	ParentID string `json:"parent_id,omitempty"`
+
+	// Annotations - object with key-value pairs that you want X-Ray to index for search
+	Annotations map[string]interface{} `json:"annotations,omitempty"`
+
+	// SubSegments contains the list of child segments
+	SubSegments []*segment `json:"subsegments,omitempty"`
+
+	// Service - optional service definition
+	Service *service `json:"service,omitempty"`
+}
+
+type service struct {
+	// Name of the service4
+	Name string `json:"name,omitempty"`
+
+	// Version - A string that identifies the version of your application that served the request.
+	Version string `json:"version,omitempty"`
+}
+
+type http struct {
```

Comment:
Unused?

Reply:
yep, this pull request was really intended for me to test the waters to see how receptive you guys were and what type of things you'll be looking for.